docs/user/rstudio/ide/guide/tools/jobs-in-action.qmd

```diff
@@ -1,49 +1,322 @@
+
+![](images/plumber-interaction.png){fig-alt="A screenshot of RStudio with the plumber-job.R file open in the Source pane, the Background job tab selected from the Console pane displaying the url to the API and it's docs, and the Viewer pane displaying the plumber API's Swagger docs."} 
+
+To interact with the API programatically in RStudio, use a R package such as [httr](https://httr.r-lib.org/) or a command line tool such as [curl](https://curl.se/). The example API has a few endpoints (`/sum`, `/echo`, `/plot`) and you can query them with `GET()` or `POST()` commands.
+
+```r
+# if needed install.packages("httr")
+```
+
+From the **Background Jobs** tab within the **Console** pane, the running API will return the URL to the API and it's rapidoc Docs:
+
+![](images/plumber-job-url.png){fig-alt="A screenshot of the Background Jobs pane, displaying the example URL to the API and the rapidoc docs."}
+
+``` bash
+# This is an example URL and 
+# it will vary between sessions
+Running plumber API at http://127.0.0.1:6285
+Running rapidoc Docs at http://127.0.0.1:6285/docs/
+```
+
+We can add specific endpoints to that URL (`http://127.0.0.1:6285`) such as `/sum` or `/echo` and then query them with the httr package.
+
+```r
+post_sum <- httr::POST("http://127.0.0.1:6285/sum", query = list(a=1, b = 2))
+
+httr::content(post_sum)
+#> [1] 3
+```
+
+```r
+get_echo <- httr::GET("http://127.0.0.1:6285/echo", query = list(msg = "dog"))
+
+httr::content(get_echo)
+#> $msg
+#> [1] "The message is: 'dog'"
+```
+
+
+## vetiver model as an API
+
+The [vetiver framework](https://vetiver.rstudio.com/) is for MLOps tasks in Python and R. The [vetiver R package](https://rstudio.github.io/vetiver-r/) has functions for deploying vetiver models as plumber APIs. This allows for remote deployment, as well as local testing in a similar way to generic plumber APIs. The vetiver example below is taken from the [vetiver 'Get Started' documentation](https://vetiver.rstudio.com/get-started/deploy.html).
+
+If necessary, install the latest versions of these packages from CRAN:
 
-![](https://solutions.rstudio.com/r/jobs/images/plumber.gif){alt=""}
+``` r
+install.packages(c("tidymodels", "vetiver", "pins", "plumber"))
+```
 
-## Extract, Transform, Load (ETL)
+First, generate a vetiver model and write out that model to a plumber API.
 
-Background and Launcher jobs are ideal for long running processes, like loading data from an external database or API.
+``` r
+library(tidymodels)
+library(vetiver)
+library(pins)
 
-![](https://solutions.rstudio.com/r/jobs/images/etl.gif){alt=""}
+car_mod <-
+    workflow(mpg ~ ., decision_tree(mode = "regression")) %>%
+    fit(mtcars)
+v <- vetiver_model(car_mod, "cars_mpg")
+model_board <- board_folder("pins-r", versioned = TRUE)
+model_board %>% vetiver_pin_write(v)
+
+# write out a vetiver model to a plumber API
+vetiver_write_plumber(model_board, "cars_mpg", file = "vetiver-plumber.R")
+```
```

Comment:
Does the above code need to be saved in a script called `vetiver-plumber.R`?

Reply:
I modified the instructions to help clarify:


> First, save the below code as `vetiver-train.R` and execute it to generate a vetiver model and write out that model to a plumber API.